Examples/MyExample(FactoryMethod)/TransportService.cs

```diff
@@ -0,0 +1,102 @@
+﻿
+namespace Examples.MyExample_FactoryMethod_
+{
+    public interface ITransport
+    {
+        void Book();
+    }
+
+    public class AirTransportationService : ITransport
+    {
+        private static AirTransportationService _airTransportationService;
+        private static readonly object _lock = new();
+
+        private AirTransportationService() { }
+
+        public static AirTransportationService GetInstance()
+        {
+            lock (_lock)
+            {
+                if (_airTransportationService == null)
+                    _airTransportationService = new AirTransportationService();
```

Comment:
it is not a good idea to mange the instance creation here. you have to move it to related factory.
SOC Principle.